resteasy-jaxrs/src/main/java/org/jboss/resteasy/core/StringParameterInjector.java

```diff
@@ -80,40 +80,39 @@ protected void initialize(Class type, Type genericType, String paramName, Class
       baseType = type;
       baseGenericType = genericType;
 
-      if (type.isArray()) baseType = type.getComponentType();
-      if (List.class.isAssignableFrom(type))
-      {
-         isCollection = true;
-         collectionType = ArrayList.class;
-      }
-      else if (SortedSet.class.isAssignableFrom(type))
-      {
-         isCollection = true;
-         collectionType = TreeSet.class;
-      }
-      else if (Set.class.isAssignableFrom(type))
+      if (type.isArray())
       {
-         isCollection = true;
-         collectionType = HashSet.class;
+    	  baseType = type.getComponentType();
       }
-      if (isCollection)
+     
+      if (!baseType.isPrimitive())
       {
-         if (genericType != null && genericType instanceof ParameterizedType)
+         paramConverter = factory.getParamConverter(baseType, baseGenericType, annotations);
+         if (paramConverter != null) 
          {
-            ParameterizedType zType = (ParameterizedType) genericType;
-            baseType = Types.getRawType(zType.getActualTypeArguments()[0]);
-            baseGenericType = zType.getActualTypeArguments()[0];

```

Comment:
re: "Right. But before your changes there was no call like the one on line 90, which uses baseGenericType before it is modified."

Yes and that's why the user registered `javax.ws.rs.ext.ParamConverterProvider` was never called with the collection genericType since when it is a collection the `baseGenericType` is modified to the data type held by the collection.

re: "The call from line 88 to 94" and "Then remove code from line 110 to 114"

In the original code the call to `factory.getParamConverter(baseType, baseGenericType, annotations);` was only invoked if `!baseType.isPrimitive()`.
In my version the first call to `factory.getParamConverter(baseType, baseGenericType, annotations);` at line 90 is done only if `!baseType.isPrimitive()` but the second one at line 110 will be invoked even if 'baseType.isPrimitive()' which is different from the original behavior when dealing with a collection.